src/pages/RepoPage/FailedTestsTab/FailedTestsPage/MetricsSection/MetricsSection.tsx

```diff
@@ -0,0 +1,451 @@
+  const selectedBranch = decodedBranch ?? testResults?.defaultBranch ?? ''
+
+  if (selectedBranch !== testResults?.defaultBranch) {
+    return null
+  }
+
+  const aggregates = testResults?.testResultsAggregates
+
+  return (
+    <>
+      <hr />
+      <div className="overflow-x-auto overflow-y-hidden md:flex">
+        <div className="mb-6 flex flex-col gap-3 border-r-2 md:mb-0">
+          <p className="pl-4 text-xs font-semibold text-ds-gray-quaternary">
+            Improve CI Run Efficiency
+          </p>
+          <div className="grid grid-cols-2">
+            <TotalTestsRunTimeCard
+              totalDuration={aggregates?.totalDuration}
+              totalDurationPercentChange={
+                aggregates?.totalDurationPercentChange
+              }
+              intervalCopy={historicalTrendToCopy(
+                queryParams?.historicalTrend as MeasurementInterval
+              )}
+            />
+            <SlowestTestsCard
+              slowestTests={aggregates?.totalSlowTests}
+              slowestTestsPercentChange={
+                aggregates?.totalSlowTestsPercentChange
+              }
+              slowestTestsDuration={aggregates?.slowestTestsDuration}
+              updateParams={updateParams}
+              isSelected={queryParams?.parameter === 'SLOWEST_TESTS'}
+            />
+          </div>
+        </div>
+        <div className="flex flex-col gap-3">
+          <p className="pl-4 text-xs font-semibold text-ds-gray-quaternary">
+            Improve Test Performance
+          </p>
+          <div
+            className={cn(
+              'grid',
+              !!flakeAggregates ? 'grid-cols-4' : 'grid-cols-2'
+            )}
+          >
+            {!!flakeAggregates ? (
+              <>
+                <TotalFlakyTestsCard
+                  flakeCount={flakeAggregates?.flakeCount}
+                  flakeCountPercentChange={
+                    flakeAggregates?.flakeCountPercentChange
+                  }
+                  updateParams={updateParams}
+                  isSelected={queryParams?.parameter === 'FLAKY_TESTS'}
+                />
+                <AverageFlakeRateCard
+                  flakeRate={flakeAggregates?.flakeRate}
+                  flakeRatePercentChange={
+                    flakeAggregates?.flakeRatePercentChange
+                  }
+                />
+              </>
+            ) : null}
+            <TotalFailuresCard
+              totalFails={aggregates?.totalFails}
+              totalFailsPercentChange={aggregates?.totalFailsPercentChange}
+              updateParams={updateParams}
+              isSelected={queryParams?.parameter === 'FAILED_TESTS'}
+            />
+            <TotalSkippedTestsCard
+              totalSkips={aggregates?.totalSkips}
+              totalSkipsPercentChange={aggregates?.totalSkipsPercentChange}
+              updateParams={updateParams}
+              isSelected={queryParams?.parameter === 'SKIPPED_TESTS'}
```

Comment:
Not blocking - would be nice to use the constant defined elsewhere (I think `TestResultsFilterParameter`)

Reply:
PR up here! https://github.com/codecov/gazebo/pull/3484